pandas-stubs/core/frame.pyi

```diff
@@ -553,20 +548,11 @@ class DataFrame(NDFrame, OpsMixin):
     def T(self) -> DataFrame: ...
     def __getattr__(self, name: str) -> Series: ...
     @overload
-    def __getitem__(  # type: ignore[misc]
-        self,
-        key: Series
-        | DataFrame
-        | Index
-        | np_ndarray_str
-        | np_ndarray_bool
-        | list[_ScalarOrTupleT]
-        | Generator[_ScalarOrTupleT, None, None],
-    ) -> DataFrame: ...
+    def __getitem__(self, key: Scalar | tuple[Hashable, ...]) -> Series: ...  # type: ignore[misc]
     @overload
-    def __getitem__(self, key: slice) -> DataFrame: ...
+    def __getitem__(self, key: Iterable[Scalar | Hashable] | slice) -> DataFrame: ...
```

Comment:
Could this be simplified to just `Iterable[Hashable]`, since anything that is `Scalar` is `Hashable` ?  (Same for `__iter__()`)